### PR TITLE
GODRIVER-2130 Update serverless tests to run against load balancer

### PR DESCRIPTION
**Pull request by kevinAlbs**
**Summary**
Update serverless tests to use a load balanced instance.

**Background & Motivation**
The Go driver already has a `test-serverless` task. This updates the task to use a load balanced instance.

[DRIVERS-1882](https://jira.mongodb.org/browse/DRIVERS-1882) describes the motivation for switching to a load balanced instance.

**Testing**
The [test-serverless](https://evergreen.mongodb.com/task/mongo_go_driver_serverless__os_ssl_40~ubuntu1804_64_go_1_16_test_serverless_patch_63855571587c4e7c7ba556410df182b3019208d6_613ec55ae3c3312a36b90949_21_09_13_03_28_27/0##comparehashes=63855571587c4e7c7ba556410df182b3019208d6&threads=maxonly) task attached to this PR runs the updated tests.

### CODE REVIEW

File: .evergreen/config.yml [hunk 1]
```diff
@@ -514,10 +514,12 @@ functions:
           export GOFLAGS=-mod=vendor
           AUTH="auth" \
           SSL="ssl" \
-          MONGODB_URI="${MONGODB_URI}" \
+          MONGODB_URI="${SINGLE_ATLASPROXY_SERVERLESS_URI}" \
           SERVERLESS="serverless" \
           SERVERLESS_ATLAS_USER="${SERVERLESS_ATLAS_USER}" \
           SERVERLESS_ATLAS_PASSWORD="${SERVERLESS_ATLAS_PASSWORD}" \
+          SINGLE_MONGOS_LB_URI="${SINGLE_ATLASPROXY_SERVERLESS_URI}" \
+          MULTI_MONGOS_LB_URI="${MULTI_ATLASPROXY_SERVERLESS_URI}" \
```
Comment (kevinAlbs) [inline .evergreen/config.yml:522]: `MULTI_MONGOS_LB_URI` connects to the load balancer fronting three Atlas Proxy processes.

`SINGLE_ATLASPROXY_SERVERLESS_URI` connects to one Atlas Proxy. It is required when a `MongoClient` needs to be pinned to one Atlas Proxy. Pinning to a server is needed when configuring a failpoint. The failpoint is handled in Atlas Proxy, so a client needs both the failpoint and the operation that triggers the failpoint to be routed to the same Atlas Proxy.

These expansions are further described in the Serverless Testing README: https://github.com/mongodb/specifications/tree/ff5d91fea80b9dcb40fb2fb3533fd1bbcee818b0/source/serverless-testing#test-runner-configuration

File: data/load-balancers/transactions.yml [hunk 2]
```diff
@@ -280,6 +282,8 @@ tests:
           - connectionCheckedInEvent: {}
 
   - description: pinned connection is released after a transient network CRUD error
+    runOnRequirements:
+      - serverless: forbid # (CLOUDP-88216) Serverless does not append error labels to errors triggered by failpoints.
```
Comment (benjirewis) [inline data/load-balancers/transactions.yml:286]: Thanks for referencing all the related CLOUDP tickets.

**Review by benjirewis — APPROVED** [review]
LGTM!

**Review by matthewdale — APPROVED** [review]
👍

**Pull request merged**